src/tools/extract-plugin.js

```diff
@@ -111,8 +124,10 @@ const processTranslateString = (path, state, funcName, types) => {
   if (args.length === 0) {
     throw path.buildCodeFrameError('Translate.string() called with no arguments');
   }
-  const msgid = processExpression(path, args[0], types);
-  const msgctxt = args[1] ? processExpression(path, args[1], types, true) : undefined;
+  const msgid = collapseWhitespace(processExpression(path, args[0], types));
+  const msgctxt = args[1]
+    ? collapseWhitespace(processExpression(path, args[1], types, true))
```

Comment:
Wouldn't it be cleaner if we `collapsedWhitespace` inside the `processExpression` and other invocations?